File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
int helper(int n,vector<int>&dp){
    if(n<0) return 0;
    if(n == 2) return 2;
    
    if(n==1) return 1;
    if(dp[n] != -1) return dp[n];


    return dp[n] = helper(n-2,dp) + helper(n-1,dp);
}
    int climbStairs(int n) {
        vector<int>dp(n+1,-1);

        //return helper(n,dp);
       if(n == 0 || n ==1 || n==2) return n;
        dp[0] =0;
        dp[1] = 1;
        dp[2] = 2;
        int ans = 0;
        for(int i =3;i<=n;i++){
          dp[i]= dp[i-2] + dp[i-1];
         // dp[n] = ans;
        }
        return dp[dp.size()-1];
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {


 
    int help(int n , int[] dp){
       if(n== 0) return 1;
       if(n < 0) return 0;
       if(dp[n]!= -1) return dp[n];
       return dp[n] = help(n-1,dp) + help(n-2,dp);
    }
    int climbStairs(int n) {
       if(n==0) return 1;
       if(n==1) return 1;
       int[] dp = new int[n+1];
       //Arrays.fill(dp,-1);
      // help(n,dp);
       dp[0] = 1;
       dp[1] = 1;
       for(int i=2;i<=n;i++){
         dp[i] = dp[i-1] + dp[i-2];
       }
       return dp[n];
    }
};